ydb/library/yql/core/type_ann/type_ann_blocks.cpp

```diff
@@ -181,6 +181,37 @@ IGraphTransformer::TStatus BlockCompressWrapper(const TExprNode::TPtr& input, TE
     return IGraphTransformer::TStatus::Ok;
 }
 
+IGraphTransformer::TStatus BlockExistsWrapper(const TExprNode::TPtr& input, TExprNode::TPtr& output, TContext& ctx) {
+    Y_UNUSED(output);
+    if (!EnsureArgsCount(*input, 1, ctx.Expr)) {
+        return IGraphTransformer::TStatus::Error;
+    }
+
+    if (!EnsureBlockOrScalarType(input->Head(), ctx.Expr)) {
+        return IGraphTransformer::TStatus::Error;
+    }
+
+    bool isScalar;
+    const TTypeAnnotationNode* blockItemType = GetBlockItemType(*input->Head().GetTypeAnn(), isScalar);
+
+    // At this point BlockItem type should be either an Optional or a Pg one.
+    // All other cases should be handled in the previous transform phases.
+    if (!EnsureOptionalType(input->Head().Pos(), *blockItemType, ctx.Expr) &&
```

Comment:
EnsureOptionalType will throw a visible error for the Pg type.

instead, check the blockItemType->GetKind() and emit an issue if both conditions are not met.

Reply:
[Fixed](https://github.com/ydb-platform/ydb/pull/3400/files#diff-87fab8180f9c448242603fa34704349a4537afcbf4a48b38a3ae398c07fa1c07R199-R204).